engine/executor/server.go

```diff
@@ -269,26 +267,18 @@ func precheckMasterMeta(
 func convertMakeTaskErrorToRPCError(
 	register registry.Registry, err error, tp frameModel.WorkerType,
 ) error {
-	if pkgClient.ErrCreateWorkerTerminate.Is(err) {
-		return rpcerror.ToGRPCError(err)
+	if errors.Is(err, errors.ErrCreateWorkerTerminate) {
+		return err
 	}
 
 	retryable, inErr := checkBusinessErrorIsRetryable(register, err, tp)
 	if inErr != nil {
 		return inErr
 	}
 	if retryable {
-		return rpcerror.ToGRPCError(
-			pkgClient.ErrCreateWorkerNonTerminate.GenWithStack(
-				&pkgClient.CreateWorkerNonTerminateError{
-					Details: err.Error(),
-				}))
-	}
-	return rpcerror.ToGRPCError(
-		pkgClient.ErrCreateWorkerTerminate.GenWithStack(
-			&pkgClient.CreateWorkerTerminateError{
-				Details: err.Error(),
-			}))
+		return errors.ErrCreateWorkerNonTerminate.Wrap(err).GenWithStackByArgs()
```

Comment:
Does `Wrap` here adds the error cause to rfc error, and only error message structure will be kept after this rfc error is converted to rpc error via rpcutil.ToGRPCError. Such as 

`[DFLOW:ErrCreateWorkerNonTerminate]create worker is not terminated: <wrapped error message>`

Reply:
Yes. The error cause will be wrapped in the string message only.

I notice that job error message was error cause before. I will add cause to the gRPC ErrorInfo's metadata, so that client can get the error cause.

Reply:
Addressed in https://github.com/pingcap/tiflow/pull/7516/commits/5ad3b7b5841d99e217b2b342b7631833241ebd5d.